eagleproject/core/blockchain.py

```diff
@@ -543,13 +546,20 @@ def maybe_store_stake_withdrawn_record(log, block):
     if len(db_staked) > 0:
         return db_staked[0]
 
+    is_updated_staked_event = log["topics"][0] == SIG_EVENT_STAKED
+    is_withdrawn_event = log["topics"][0] == SIG_EVENT_WITHDRAWN
+    is_staked_event = log["topics"][0] == DEPRECATED_SIG_EVENT_STAKED or is_updated_staked_event
+
     staked = OgnStaked(
         tx_hash=tx_hash,
         log_index=log_index,
         block_time=block.block_time,
         user_address="0x" + log["topics"][1][-40:],
-        is_staked=log["topics"][0] == SIG_EVENT_STAKED,
+        is_staked=is_staked_event,
         amount=int(_slot(log["data"], 0), 16) / 1e18,
+        staked_amount=int(_slot(log["data"], 1), 16) / 1e18 if is_withdrawn_event else 0,
```

Comment:
Isn't this the same as using `amount` and `is_staked`?  What's gained by storing this?

Reply:
ah, I guess you're going for having the Stake/Withdraw amounts line up and not be negative when all is said and done.  I might suggest a different name for clarity but I see what you're going for now.  n/m